### PR TITLE
EN6 53 - CMS ContentSettings API Endpoint

### PR DESCRIPTION
**Pull request by ffleandro**
Adds the endpoint for retrieving and editing ContentSettings.

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/pom.xml [hunk 1]
```diff
@@ -189,5 +189,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.2</version>
+        </dependency>
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/pom.xml]: How thoroughly has this been tested? We tried adding lombok to the core at some point in the past and it broke some things so we had to role it back.

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/contentsettings/ContentSettingsService.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.plugins.jacms.aps.system.services.contentsettings;
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/contentsettings/ContentSettingsService.java]: This class should have unit test coverage

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentsettings/ContentSettingsController.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.plugins.jacms.web.contentsettings;
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentsettings/ContentSettingsController.java]: Add unit test coverage

File: plugins/entando-plugin-jacms/pom.xml (hunk 1 above)
Comment (ffleandro) [reply plugins/entando-plugin-jacms/pom.xml]: Used lombok on the survey plugin without any issues.
Ran all the tests on the cms plugin and everything ran without any issues.

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentsettings/ContentSettingsController.java (hunk 3 above)
Comment (ffleandro) [reply plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/web/contentsettings/ContentSettingsController.java]: As we talked, will only do integration tests.

**Comment by entando-service** [comment]
All is Well

**Pull request merged**